File: testdata/fixtures/bench/csharp/vulnerable/a03_command_injection.cs
// Source: CWE-78 - OS Command Injection via Process.Start in C#
// Expected: BATOU-INJ, TAINT
// OWASP: A03:2021 - Injection (Command Injection)

using System;
using System.Diagnostics;

namespace App
{
    class CommandRunner
    {
        public void RunCommand(string userInput)
        {
            var cmd = "ping -c 1 " + userInput;
            Process.Start("/bin/sh", "-c " + cmd);
        }

        static void Main(string[] args)
        {
            var runner = new CommandRunner();
            runner.RunCommand(args[0]);
        }
    }
}

File: testdata/fixtures/bench/swift/vulnerable/a03_command_injection.swift
// Source: CWE-78 - OS Command Injection via Process in Swift
// Expected: BATOU-INJ, TAINT
// OWASP: A03:2021 - Injection (Command Injection)

import Foundation

func runCommand(userInput: String) {
    let task = Process()
    task.executableURL = URL(fileURLWithPath: "/bin/sh")
    task.arguments = ["-c", "ping -c 1 " + userInput]
    try? task.run()
    task.waitUntilExit()
}

let args = CommandLine.arguments
if args.count > 1 {
    runCommand(userInput: args[1])
}
